akka-cluster-sharding/src/test/scala/akka/cluster/sharding/InactiveEntityPassivationSpec.scala

```diff
@@ -47,54 +54,71 @@ object InactiveEntityPassivationSpec {
   val extractShardId: ShardRegion.ExtractShardId = {
     case msg: Int => (msg % 10).toString
   }
-
 }
 
-class InactiveEntityPassivationSpec extends AkkaSpec(InactiveEntityPassivationSpec.config) {
+abstract class AbstractInactiveEntityPassivationSpec(c: Config) extends AkkaSpec(c) {
   import InactiveEntityPassivationSpec._
 
-  val smallTolerance = 300.millis
+  private val smallTolerance = 300.millis
+
+  def start(probe: TestProbe): ActorRef = {
+    // single node cluster
+    Cluster(system).join(Cluster(system).selfAddress)
+    val settings = ClusterShardingSettings(system)
+    ClusterSharding(system).start(
+      "myType",
+      InactiveEntityPassivationSpec.Entity.props(probe.ref),
+      settings,
+      extractEntityId,
+      extractShardId,
+      ClusterSharding(system).defaultShardAllocationStrategy(settings),
+      Passivate)
+  }
 
-  "Passivation of inactive entities" must {
+  def timeUntilPassivate(region: ActorRef, probe: TestProbe): FiniteDuration = {
+    region ! 1
+    region ! 2
+    val responses = Set(probe.expectMsgType[GotIt], probe.expectMsgType[GotIt])
+    responses.map(_.id) should ===(Set("1", "2"))
+    val timeOneSawMessage = responses.find(_.id == "1").get.when
+    Thread.sleep(1000)
+    region ! 2
+    probe.expectMsgType[GotIt].id should ===("2")
+    Thread.sleep(1000)
+    region ! 2
+    probe.expectMsgType[GotIt].id should ===("2")
+
+    val timeSinceOneSawAMessage = (System.nanoTime() - timeOneSawMessage).nanos
+    (3.seconds - timeSinceOneSawAMessage) - smallTolerance
```

Comment:
this seems strange, `timeSinceOneSawMessage` is a `System.nanoTime` timestamp, that can't be subtracted from a Duration. Maybe this was intended as:
```
(System.nanoTime - timeSinceOneSawAMessage).nanos + smallTolerance
```
?

Reply:
Seemed curious to me as well. Updating.

Reply:
@patriknw wdyt:
```scala
(settings.passivateIdleEntityAfter.fromNow.timeLeft - timeOneSawMessage.nanos) + smallTolerance
```

Reply:
`timeOneSawMessage.nanos` is not correct, because timeOneSawMessage is a nanoTime timestamp, not a duration.

but wait, the original is almost correct, I misread timeSinceOneSawAMessage and timeOneSawMessage

should probably be `+ smallTolerance` instead of `-`, and use `settings.passivateIdleEntityAfter` instead of `3.seconds`

Reply:
@patriknw updated, indeed we need the `+` but I removed what seemed an unnecessary sending of `$id + passivate` message by the test entity actor to the probe on receive on `Passivate` and it now just asserts no message received which seems clearer.